### PR TITLE
[Core] Mortar utilities missing #pragma omp atomic

### LINKED ISSUE #3468: [Core] MortarMapper test failing
~~~
======================================================================
FAIL: test_mortar_mapping_quad (test_mortar_mapper.TestMortarMapperCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/philippb/software/Kratos_Master/kratos/tests/test_mortar_mapper.py", line 164, in test_mortar_mapping_quad
    self._mapper_tests(input_filename, 4, 4)
  File "/home/philippb/software/Kratos_Master/kratos/tests/test_mortar_mapper.py", line 122, in _mapper_tests
    check.ExecuteFinalizeSolutionStep()
  File "/home/philippb/software/Kratos_Master/kratos/python_scripts/from_json_check_result_process.py", line 154, in ExecuteFinalizeSolutionStep
    self.assertTrue(isclosethis, msg=(str(value) + " != " + str(value_json) + ", rel_tol = " + str(reltol) + ", abs_tol = " + str(tol) + " : Error checking node " + str(node.Id) + " " + variable_name + " results."))
AssertionError: False is not true : 10.04886216959597 != 10.043189538154905, rel_tol = 1e-06, abs_tol = 0.001 : Error checking node 242 TEMPERATURE results.

======================================================================
FAIL: test_mortar_mapping_quad_tri (test_mortar_mapper.TestMortarMapperCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/philippb/software/Kratos_Master/kratos/tests/test_mortar_mapper.py", line 168, in test_mortar_mapping_quad_tri
    self._mapper_tests(input_filename, 4, 3)
  File "/home/philippb/software/Kratos_Master/kratos/tests/test_mortar_mapper.py", line 122, in _mapper_tests
    check.ExecuteFinalizeSolutionStep()
  File "/home/philippb/software/Kratos_Master/kratos/python_scripts/from_json_check_result_process.py", line 162, in ExecuteFinalizeSolutionStep
    self.assertTrue(isclosethis, msg=(str(value) + " != " + str(value_json) + ", rel_tol = " + str(reltol) + ", abs_tol = " + str(tol) + " : Error checking node " + str(node.Id) + " " + variable_name + " results."))
AssertionError: False is not true : [3](45.8241,40.0216,3.90729) != 45.820688332486334, rel_tol = 1e-06, abs_tol = 0.001 : Error checking node 744 DISPLACEMENT results.

----------------------------------------------------------------------
~~~
GCC 5.4

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #3468

### CODE REVIEW

**Comment by philbucher** [comment]
7aaa5548-5e81-4b07-b087-f964bbe6b47d

**Comment by philbucher** [comment]
Will take a look tmr ;)

**Comment by loumalouomega** [comment]
> 7aaa5548-5e81-4b07-b087-f964bbe6b47d

You got me :P

File: kratos/utilities/mortar_utilities.h [hunk 1]
```diff
@@ -1087,67 +1108,72 @@ inline void MortarUtilities::AddAreaWeightedNodalValue<Variable<array_1d<double,
     if (null_area) KRATOS_WARNING("WARNING:: NODE OF NULL AREA.") << " ID: " << pThisNode->Id() << std::endl;
```
Comment (philbucher) [inline kratos/utilities/mortar_utilities.h:1108]: are you sure this should only be issued in debug?

**Comment by philbucher** [comment]
still fails, even with one thread
maybe you have to update the results?

**Comment by loumalouomega** [comment]
I run it 100 times, and it doesn't fail to me, I will, try in full debug

**Comment by loumalouomega** [comment]
@philbucher I run 100 times on full debug, no error at all, are you sure?

**Comment by loumalouomega** [comment]
@philbucher are you sure you still have that error?, I run 100 both in release and full debug and I don't have any problem. Anyway this already solves some OMP errors.

**Comment by philbucher** [comment]
I will try again later, meanwhile you can merge this since it solves some errors

**Review by philbucher — APPROVED** [review]

**Pull request merged**